xclim/sdba/base.py

```diff
@@ -366,23 +403,198 @@ def parse_group(func: Callable) -> Callable:
 
     Adds the possiblity to pass a window argument and a list of dimensions in group.
     """
-    default_group = signature(func).parameters["group"].default
+    sig = signature(func)
+    if "group" in sig.parameters:
+        default_group = sig.parameters["group"].default
+    else:
+        default_group = None
 
     @wraps(func)
     def _parse_group(*args, **kwargs):
-        group = kwargs.setdefault("group", default_group)
-        if not isinstance(group, Grouper):
-            if not isinstance(group, str):
-                dim, *add_dims = group
-            else:
-                dim = group
-                add_dims = []
-            kwargs["group"] = Grouper(
-                group=dim,
-                window=kwargs.pop("window", 1),
-                add_dims=add_dims,
-                interp=kwargs.get("interp", False),
-            )
+        if default_group:
+            kwargs.setdefault("group", default_group)
+        elif "group" not in kwargs:
+            raise ValueError("'group' argument not given.")
+        if not isinstance(kwargs["group"], Grouper):
+            kwargs = Grouper.from_kwargs(**kwargs)
         return func(*args, **kwargs)
 
     return _parse_group
+
+
+def duck_empty(dims, sizes, chunks=None):
+    """Return an empty DataArray based on a numpy or dask backend, depending on the chunks argument."""
+    shape = [sizes[dim] for dim in dims]
+    if chunks:
+        chnks = [chunks.get(dim, (sizes[dim],)) for dim in dims]
+        content = dsk.empty(shape, chunks=chnks)
+    else:
+        content = np.empty(shape)
+    return xr.DataArray(content, dims=dims)
+
+
+def map_blocks(reduces=None, **outvars):
+    """
+    Decorator for declaring functions and wrapping them into a map_blocks. It takes care of constructing
+    the template dataset.
+
+    If `group` in the kwargs, it is assumed that `group.dim` is the only dimension reduced or modified,
+    and that some other dimensions might be added, but no other existing dimension will be modified.
+
+    Arguments to the decorator are mappings from variable name in the output to its *new* dimensions.
+    Dimension order is not preserved.
+    The placeholders "<PROP>" and "<DIM>" can be used to signify `group.prop` and `group.dim` respectively.
```

Comment:
This `<PROP>` and `<DIM>` thing doesn't feel pythonic. I assume there weren't other options?

Reply:
There are other options I guess. I could hide this in constants like `ADDITIVE`, so the decorator looks like: 
```
@map_blocks(reduces=[sdba.PROP, "quantiles"], scen=[])
```

We need a simple flag. The decorator needs to be understandable by the developper and the flag to be unambiguous for the code.  It is never used outside the definition of sdba algorithms.

Reply:
@huard  Would like your opinion on the new syntax. I made class-constants `Grouper.PROP` and `Grouper.DIM`, that are used the same way as `<PROP>` and `<DIM>`. It does feel a bit more pythonic.